lib/azure/armrest/armrest_service.rb

```diff
@@ -161,7 +161,11 @@ class << self
         private
 
         def rest_execute(options, http_method = :get)
-          RestClient::Request.execute(options.merge(:method => http_method))
+          options = options.merge(
+            :method => http_method,
+            :url    => Addressable::URI.parse(options[:url]).normalize.to_str
```

Comment:
@djberg96 I actually don't know. I just followed what is said in  http://stackoverflow.com/questions/20254749/restclient-parsing-uri-with-space-on-request-path that "The documentation suggests to parse and normalize the URL first".
I don't know what document it referenced 😄